app/commands/git/sync.rb

```diff
@@ -0,0 +1,72 @@
+module Git
+  class Sync
+    include Mandate
+    initialize_with :track, :synced_to_git_sha
+
+    def call
+      sync! unless synced_to_head?
+    end
+
+    def sync!
+      raise NotImplementedError
+    end
+
+    def update_git_repo!
+      git_repo.update!
+    end
+
+    def filepath_in_diff?(filepath)
+      diff.each_delta.any? do |delta|
+        [delta.old_file[:path], delta.new_file[:path]].include?(filepath)
+      end
+    end
+
+    memoize
+    def git_repo
+      Git::Repository.new(track.slug, repo_url: track.repo_url)
+    end
+
+    memoize
+    def head_git_track
+      Git::Track.new(track.slug, git_repo.head_sha, repo: git_repo)
+    end
+
+    memoize
+    def synced_git_track
+      Git::Track.new(track.slug, synced_to_git_sha, repo: git_repo)
+    end
+
+    memoize
+    def synced_to_head?
+      synced_git_track.commit.oid == head_git_track.commit.oid
+    end
+
+    memoize
+    def track_config_modified?
+      filepath_in_diff?(head_git_track.config_filepath)
+    end
+
+    memoize
+    def diff
+      head_git_track.commit.diff(synced_git_track.commit)
+    end
+
+    memoize
+    def config_concept_exercises
```

Comment:
How about we switch the `config` on these to be a suffix?
```suggestion
    def concept_exercises_config
```